### PR TITLE
Update voter section of best practices

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | yes |
| Applies to | 2.8+ |
| Fixed tickets | https://github.com/symfony/symfony-docs/issues/4389, https://github.com/symfony/symfony-docs/issues/5728 |

### CODE REVIEW

File: best_practices/security.rst [hunk 1]
```diff
@@ -264,37 +264,66 @@ the same ``getAuthorEmail`` logic you used above:
...
 
-        protected function getSupportedClasses()
+        protected function supports($attribute, $subject)
         {
-            return array('AppBundle\Entity\Post');
+            if (!in_array($attribute, array(self::CREATE, self::EDIT))) {
+                return false;
+            }
+
+            if (!$subject instanceof Post) {
```
Comment (xabbuh) [inline best_practices/security.rst]: We need to test first if `$subject` really is an object.

File: best_practices/security.rst [hunk 2]
```diff
@@ -264,37 +264,66 @@ the same ``getAuthorEmail`` logic you used above:
...
         }
 
-        protected function isGranted($attribute, $post, $user = null)
+        protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
         {
+            $user = $token->getUser();
+            /** @var Post */
+            $post = $subject; // $subject must be a Post instance, thanks to the supports method
+
             if (!$user instanceof UserInterface) {
```
Comment (xabbuh) [inline best_practices/security.rst]: same here (for `$user`)

File: best_practices/security.rst (hunk 1 above)
Comment (WouterJ) [inline best_practices/security.rst]: That's not needed, instanceof with a non-object simply evaluates in `false`: https://3v4l.org/FJqoI

**Comment by xabbuh** [comment]
:+1:

**Comment by weaverryan** [comment]
So awesome :). Thanks for the help Wouter!

**Pull request merged**